sound/soc/sof/intel/hda-stream.c

```diff
@@ -193,23 +202,39 @@ int hda_dsp_stream_put(struct snd_sof_dev *sdev, int direction, int stream_tag)
 {
 	struct hdac_bus *bus = sof_to_bus(sdev);
 	struct hdac_stream *s;
+	bool active_capture_stream = false;
+	bool found = false;
 
 	spin_lock_irq(&bus->reg_lock);
 
-	/* find used stream */
+	/*
+	 * close stream matching the stream tag
+	 * and check if there are any open capture streams.
+	 */
 	list_for_each_entry(s, &bus->stream_list, list) {
```

Comment:
optional: maybe it would be even clearer to write this as
```
list_for_...() {
	if (!s->opened)
		continue;
	if (s->direction == direction && s->stream_tag == stream_tag) {
		...
	} else if (s->direction == SNDRV_PCM_STREAM_CAPTURE) {
		...
	}
}
```
but I'd say this is already nitpicking. Also, personally I'd also remove curly braces from the `list_for...()` loop but I know that many have a different opinion on this too.